.github/workflows/ci.yml

```diff
@@ -51,3 +52,38 @@ jobs:
       - run: cabal build
       - run: cabal test
       - run: ./script/haddock
+
+      - name: Create folder for artifacts
+        id: artifact
+        run: |
+          mkdir artifact
+          mkdir artifact/${{ matrix.os }}
+          mkdir artifact/${{ matrix.os }}/${{ matrix.ghc }}
+          echo ::set-output name=directory::'artifact/${{ matrix.os }}/${{ matrix.ghc }}'
+
+      - name: Generate package ready for Hackage distribution
+        run: cabal sdist --output-dir ${{ steps.artifact.outputs.directory }}
+
+      - name: Upload the artifact to GitHub
```

Comment:
@DavidMazarro actually I just realize that [tfausak/strive](https://github.com/tfausak/strive/blob/main/.github/workflows/ci.yaml) calls `cabal update` directly in the context of `release` job without calling `haskell/actions/setup` first, so my guess is that `ubuntu-latest` runner already has `cabal` pre-installed? If that is the case, I think we could get rid of the whole upload-artifact/download-artifact thing and call `cabal sdist` and `cabal upload` directly at the `release` job:

```yml
release:
  ...
  steps:
    - uses: actions/checkout@v3
    - run: cabal sdist
    - run: cabal upload ...
```